docs/readmes/proposals/p010_vendor_neutral_dp.md

```diff
@@ -416,7 +456,250 @@ scope of work.
+
+
+# Compatibility
+
+Domain Proxy solution will reuse as much of the Magma services
+
+It should be easily deployed within existing Magma environment.
+
+As DP will be implementing radio intrefaces, the intention is to follow
+and comply with the Magma Community guidelines and codebase where applicable.
+
+# Observability and Debug
+
+The plan is to follow Magma best practises regarding application observability
+(logging, telemetry, debugging applications and the DP state)
+as well as utilize already existing tools and solutions in Magma infrastructure.
+
+# Implementation
+
+## System Architecture
+
+The following picture shows the DP components and the communication layout of the solution.
+
+<img src="media/p010_vendor_neutral_dp_image3.png" />
+
+---
+
+## DP Microservices
+
+* **Protocol Controller (PC)**
+  
+  Protocol Controller is responsible for receiving the initial traffic
+  from individual CBSDs (radios), translating the requests
+  into a common API protocol and forwarding them to the Radio Controller.
+
+  On the way back from SAS the requests will be put back into the PC
+  and distributed across CBSDs that sent them.
+
+  There will be one PC for each protocol that the DP needs to handle.
+
+  Initially this means only Domain Proxy protocol, but the idea behind
+  Protocol Controllers is for them to be written in such a manner that
+  having one of them in place, it should be straightforward to add
+  another one (it can be achieved by having a ProtocolController abstract class
+  to provide structure for particular PC implementations).
+  
+  To scale the system better we are decoupling PC (responsible for protocol)
+  from Radio controller (responsible for the business domain).
+  Additional benefit is maintaining all of the protocol customizations
+  and tricks at the PC level and making the PC layer scalable
+  according to the number of eNBs of each type.
+
+* **Radio Controller (RC)**
+  
+  RC is responsible for the CBSD to SAS communication
+  abstracted from vendor specific protocols.
+  
+  RC on one hand is responsible for placing the requests coming from PCs in the queue
+  and on the other for segregating the responses coming from Configuration Controller
+  and sending them back to the appropriate PC.
+
+* **Configuration Controller (CC)**
+
+  Configuration Controller is responsible for polling requests periodically from the queue,
+  aggregating them into one request in format understandable by SAS,
+  posting them over to SAS
+  and once a response is received, passing it on to RC.
+
+* **Message Queue**
+
+  RabbitMQ instance for storage of requests received by the RC
+  allowing them to be polled by a consumer.
+  Requests for which acknowledgement is received upon polling
+  will be deleted from the queue,
+  while the ones without acknowledgement will be requeued.
+
+* **Database (optional, to be implemented at stage 3)**
+
+  Relational DB storing the state of the DP.
```

Comment:
@karthiksubraveti 
We did mention that our services will be stateless Python applications. And I think the stateless attribute still holds on per-service basis, because the services alone will not have any state. But the entire Domain Proxy will in the next development stages in the Active mode.

Our DP will be implementing two modes:
1. Passive mode is just a passthrough of CBSD requests to SAS. In this mode there is no need to preserve any sort of state. Please note that this mode will still add value in form of request batching.

2. Active mode - Domain Proxy, in Active Mode, will be implementing states of the CBSDs. In this mode, the current CBSD request state as well as SAS response state have to be stored in the Database.